source/extensions/geoip_providers/maxmind/geoip_provider.cc

```diff
@@ -116,12 +125,48 @@ void GeoipProvider::lookup(Geolocation::LookupRequest&& request,
                            Geolocation::LookupGeoHeadersCallback&& cb) const {
   auto& remote_address = request.remoteAddress();
   auto lookup_result = absl::flat_hash_map<std::string, std::string>{};
-  lookupInCityDb(remote_address, lookup_result);
+  // lookup countryDb only when city_db_path_ is not configured
```

Comment:
by seeing this comment, should we do `oneof` in proto so we only allow either country or city map?

Reply:
makes sense, let me update the PR to try `oneof`

Reply:
An `oneof` change of the existing field may break the backward compatibility, you should not change it and add configuration time exception to notice the user setting both country and city databases is not valid.

Reply:
since we are adding a new field `country_db_path_` in this PR, and users will only be able to pick `country_db_path_` or `city_db_path` after this change, i.e., currently they would either configure `city_db_path_` or not using it, would we still have backward compatibility issues of we can adopt `oneof` directly in proto definition in this PR?

Reply:
got it, thank you both for the context. will update the PR to add the config time exception